etna/insights/models.py

```diff
@@ -30,14 +32,25 @@ def get_context(self, request):
 
     content_panels = BasePage.content_panels + [
         FieldPanel("sub_heading"),
-        PageChooserPanel("featured_insight"),
         StreamFieldPanel("body"),
     ]
     promote_panels = BasePage.promote_panels + TeaserImageMixin.promote_panels
 
     subpage_types = ["insights.InsightsPage"]
 
 
+@register_snippet
+class InsightsPageTag(TaggedItemBase):
+    content_object = ParentalKey(
+        "insights.InsightsPage", on_delete=models.CASCADE, related_name="tagged_items"
+    )
+    free_tagging = False
+
+    class Meta:
+        verbose_name = "name"
+        verbose_name_plural = "slug"
+
+
```

Comment:
You've defined a new 'through model' here, but I was expecting there to also be a custom `Tag` subclass too, [as [demonstrated  in the example here](https://docs.wagtail.org/en/stable/reference/pages/model_recipes.html#custom-tag-models). It is the custom `Tag` model that should have `free_tagging = False` applied to it to disable the free tagging, rather than the through model.

Reply:
This is now in a new custom model InsightsTag.